### PR TITLE
Improve code & book documentation

### PR DESCRIPTION
**Pull request by volhovm**
This is the part with some small refactorings I was doing while working on the vinegar RFC. No code logic change intended, changes should be purely cosmetic.

Review by commit.

### CODE REVIEW

File: kimchi/src/proof.rs [hunk 1]
```diff
@@ -735,85 +742,83 @@ pub mod caml {
...
+                    .map(|x| x.map(&|x| x.into_iter().map(Into::into).collect())),
+                range_check_lookup_selector: pe
+                    .range_check_lookup_selector
+                    .map(|x| x.map(&|x| x.into_iter().map(Into::into).collect())),
+                foreign_field_mul_lookup_selector: pe
+                    .foreign_field_mul_lookup_selector
+                    .map(|x| x.map(&|x| x.into_iter().map(Into::into).collect())),
+            };
+
+            (first, second)
```
Comment (querolita) [inline kimchi/src/proof.rs:821]: Not sure I like this distinction

File: kimchi/src/prover.rs [hunk 2]
```diff
@@ -1101,7 +1101,7 @@ where
 
         //~ 1. Evaluate the same polynomials without chunking them
         //~    (so that each polynomial should correspond to a single value this time).
-        let evals = {
+        let evals: ProofEvaluations<PointEvaluations<G::ScalarField>> = {
```
Comment (querolita) [inline kimchi/src/prover.rs:1104]: Why is adding the explicit type needed now?

File: poly-commitment/src/evaluation_proof.rs [hunk 3]
```diff
@@ -419,6 +446,7 @@ pub struct OpeningProof<G: AffineCurve> {
     pub z1: G::ScalarField,
     #[serde_as(as = "o1_utils::serialization::SerdeAs")]
     pub z2: G::ScalarField,
+    /// A final folded commitment base
```
Comment (querolita) [inline poly-commitment/src/evaluation_proof.rs:449]: Folded?

File: kimchi/src/proof.rs (hunk 1 above)
Comment (volhovm) [reply kimchi/src/proof.rs:821]: Let me explain: 
1. The original function's `return` statement was `80` lines, including nested expressions and a lot of commas, so it was very confusing as to what is the return type at all. I like it when you can fast-forward to the end of the function and see what it does.
2. The `From::` conversions in this whole function are, arguably, quite non-intuitive -- it took me quite some time to understand how `map` (being a custom function! and not a generic iterator) is implemented. Figuring this out is further complicated by `From::` instances being picked implicitly, so it takes time to find them. This is why I explicitly typed `first` so that one can see by example what's happening here.

TLDR I just spent 15 minutes trying to figure out what's going on here and thought this might help...

File: kimchi/src/prover.rs (hunk 2 above)
Comment (volhovm) [reply kimchi/src/prover.rs:1104]: I guess readability. `evals` sounds like it can be any evaluations -- `ProofEvaluations<G::ScalarField>` or `PointEvaluations<somethingelse>`. Ultimately I don't know -- I probably added it because I found it confusing... But I don't think explicit typing hurts here, does it?

File: poly-commitment/src/evaluation_proof.rs (hunk 3 above)
Comment (volhovm) [reply poly-commitment/src/evaluation_proof.rs:449]: Yeah, but not in the instance-folding way. At each (1/log N) step of bulletproofs we have several vectors that get "folded in half", so that's what I meant. I can say "log-folded" or "IPA-folded" if this is more clear?

Comment (querolita) [reply poly-commitment/src/evaluation_proof.rs:449]: ah yeah, much more

**Review by querolita — APPROVED** [review]
